KarhooUISDK/UIComponents/AddressBarMVP/PrebookFieldMVP/KarhooPrebookFieldView.swift

```diff
@@ -37,44 +48,109 @@ public final class KarhooPrebookFieldView: UIView {
+            closeButton.heightAnchor.constraint(equalToConstant: Constants.closeButtonSize),
+            closeButton.leadingAnchor.constraint(equalTo: dateTimeStackContainer.trailingAnchor),
+            closeButton.topAnchor.constraint(lessThanOrEqualTo: dateTimeView.topAnchor),
+            closeButton.trailingAnchor.constraint(equalTo: dateTimeView.trailingAnchor),
+            closeButton.bottomAnchor.constraint(lessThanOrEqualTo: dateTimeView.bottomAnchor)
+        ].forEach { $0.isActive = true }
+
+        [
+            prebookButton.widthAnchor.constraint(equalToConstant: Constants.preBookButtonSize),
+            prebookButton.heightAnchor.constraint(equalToConstant: Constants.preBookButtonSize)
+        ].forEach { $0.isActive = true }
         
+        [
+            dateTimeStackContainer.leadingAnchor.constraint(equalTo: dateTimeView.leadingAnchor, constant: 5.0),
+            dateTimeStackContainer.centerYAnchor.constraint(equalTo: closeButton.centerYAnchor)
+        ]
+            .forEach { $0.isActive = true }
+    }
+
+    private func setupStackView() {
         stackContainer = UIStackView()
         stackContainer.accessibilityIdentifier = "stackView"
         stackContainer.translatesAutoresizingMaskIntoConstraints = false
         stackContainer.axis = .horizontal
-        addSubview(stackContainer)
-        
+    }
+    
+    private func setupDateTimeView() {
         dateTimeView = UIView()
         dateTimeView.accessibilityIdentifier = KHPrebookFieldID.dateTimeView
         dateTimeView.translatesAutoresizingMaskIntoConstraints = false
         dateTimeView.isHidden = true
-        stackContainer.addArrangedSubview(dateTimeView)
         
+    }
+    
+    private func setupDateTimeStackView() {
         dateTimeStackContainer = UIStackView()
         dateTimeStackContainer.accessibilityIdentifier = "dateTime_stackView"
         dateTimeStackContainer.translatesAutoresizingMaskIntoConstraints = false
         dateTimeStackContainer.axis = .vertical
-        addSubview(dateTimeStackContainer)
-        
-        timeLabel = UILabel()
-        timeLabel.accessibilityIdentifier = KHPrebookFieldID.timeLabel
-        timeLabel.translatesAutoresizingMaskIntoConstraints = false
-        timeLabel.textAlignment = .center
-        timeLabel.font = KarhooUI.fonts.footnoteBold()
-        timeLabel.textColor = KarhooUI.colors.darkGrey
-        dateTimeStackContainer.addArrangedSubview(timeLabel)
-        
-        dateLabel = UILabel()
-        dateLabel.accessibilityIdentifier = KHPrebookFieldID.dateLabel
-        dateLabel.translatesAutoresizingMaskIntoConstraints = false
-        dateLabel.textAlignment = .center
-        dateLabel.font = KarhooUI.fonts.footnoteRegular()
-        dateLabel.textColor = KarhooUI.colors.darkGrey
-        dateTimeStackContainer.addArrangedSubview(dateLabel)
-        
+    }
+    
+    private func setupTimeLabel() {
```

Comment:
Instead of having a method for each control, could you instantiate them as lazy vars? See the `KarhooCheckoutViewController` as an example. This goes for all controls